src/web/app/pages-student/student-home-page/student-home-page.component.ts

```diff
@@ -94,22 +94,24 @@ export class StudentHomePageComponent implements OnInit {
             const sortedFss: FeedbackSession[] = this.sortFeedbackSessions(fss);
 
             const studentSessions: StudentSession[] = [];
-            for (const fs of sortedFss) {
-              const isOpened: boolean = fs.submissionStatus === FeedbackSessionSubmissionStatus.OPEN;
-              const isWaitingToOpen: boolean =
-                fs.submissionStatus === FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN;
-              const isPublished: boolean = fs.publishStatus === FeedbackSessionPublishStatus.PUBLISHED;
-              this.feedbackSessionsService.hasStudentResponseForFeedbackSession(course.courseId,
-                fs.feedbackSessionName)
-                .subscribe((hasRes: HasResponses) => {
-                  const isSubmitted: boolean = hasRes.hasResponses;
+            this.feedbackSessionsService.hasStudentResponseForAllFeedbackSessionsInCourse(course.courseId)
+              .subscribe((hasRes: HasResponses) => {
+                for (const fs of sortedFss) {
+                  const isOpened: boolean = fs.submissionStatus === FeedbackSessionSubmissionStatus.OPEN;
+                  const isWaitingToOpen: boolean =
+                    fs.submissionStatus === FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN;
+                  const isPublished: boolean = fs.publishStatus === FeedbackSessionPublishStatus.PUBLISHED;
+                  const isSubmitted: boolean = hasRes.hasResponsesBySession[fs.feedbackSessionName];
+                  if (isSubmitted === undefined) {
```

Comment:
on the topic of implementation, when would this be `undefined`? I would expect all the sessions I see in the table to have a key here.

Reply:
It's merely a defensive programming pattern I put here.